### PR TITLE
added warmup

### PR DESCRIPTION
**Pull request by AnastasiiaYaitska**

### CODE REVIEW

File: nodejs/week2/warmup/app.js [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+  res.send("This is a search engine");
+});
+
+// GET /search
+app.get("/search", (req, res) => {
+  const q = req.query.q;
+
+  try {
+    if (q) {
+      const filteredDoc = documents.filter((doc) => {
```
Comment (te-online) [inline nodejs/week2/warmup/app.js]: The variable name `filteredDoc` indicates you're assigning just a single document to it, but in reality it's an array, right? What if we would find a better name for the variable?

File: nodejs/week2/warmup/app.js [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+          (value) =>
+            typeof value === "string" &&
+            value.toLocaleLowerCase().includes(q.toLocaleLowerCase())
+        );
+      });
+
+      if (filteredDoc) {
+        res.send(filteredDoc);
+      } else {
+        res.status(404).send("Document not found");
```
Comment (te-online) [inline nodejs/week2/warmup/app.js]: In context of the search always returning an array, maybe we could find a better error message instead?

File: nodejs/week2/warmup/app.js [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+});
+
+// POST /search
+app.post("/search", (req, res) => {
+  try {
+    const q = req.query.q;
+    const fields = req.body.fields;
+
+    if (q && fields) {
+      res.status(400).send("Query parameter and data can't be provided");
```
Comment (te-online) [inline nodejs/week2/warmup/app.js]: Should we be able to combine `q` and `fields`? Why did you decide to make them exclusive?

File: nodejs/week2/warmup/app.js [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+    } else if (fields) {
+      const document = documents.filter((doc) => {
+        const docKeyValue = Object.entries(doc);
+        return docKeyValue.some(([key, value]) => {
+          return doc[key] === fields[key] && doc[value] === fields[value];
```
Comment (te-online) [inline nodejs/week2/warmup/app.js]: Your `q` search is case-insensitive, but the `fields` search has to match exactly.

- If you want to match exactly the field, why not use `doc[fields[key]] === fields[value]`?
- If you want it to be case-insensitive, you'd have to add that functionality

File: nodejs/week2/warmup/app.js [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+        });
+      });
+
+      if (document) {
```
Comment (te-online) [inline nodejs/week2/warmup/app.js]: Also here, the naming could be improved 🙂

File: nodejs/week2/warmup/app.js [hunk 6]
```diff
@@ -0,0 +1,91 @@
...
+        res.send(document);
```
Comment (te-online) [inline nodejs/week2/warmup/app.js]: It doesn't look like you're taking the `q` value into account at all? 🤓

**Review by te-online — CHANGES_REQUESTED** [review]
Good work 👍 

- ✅ You created all the endpoints that were required
- ✅ You added case-insensitive search by using the `q` parameter
- ⚠️ You didn't allow `q` and `fields` to be provided at the same time
- ⚠️ Your fields search algorithm could either be simplified or made case-insensitive
- 💅 You could improve variable naming

File: nodejs/week2/warmup/app.js (hunk 6 above)
Comment (AnastasiiaYaitska) [reply nodejs/week2/warmup/app.js]: Thank you Thomas!! ☺️ Can you take a look now?

Comment (AnastasiiaYaitska) [reply nodejs/week2/warmup/app.js]: Thank you Thomas!!☺️ Canyou takea. look now?

Comment (te-online) [reply nodejs/week2/warmup/app.js]: I think you addressed all comments, but this one you skipped? 😊 `q` is still not being used in the code for this route, as far as I can see?

**Pull request closed without merging**